communication/nats/connector_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package nats
+
+import (
+	"testing"
+	"time"
+
+	"github.com/nats-io/go-nats"
+	"github.com/nats-io/nats-server/server"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBrokerConnector(t *testing.T) {
```

Comment:
1. Would be better to have isolated unit test which doesn't start real nats server. Calling connector.Connect could assert that mock conn Open was called and is added to registry map. Also it could assert that after closing conn it's removed from registry.
2. ReconnectAll is not tested. Again if mock conn is used it's enough to test if Check method is called.

Reply:
How is testing with mocks better than a proper unit test?